### PR TITLE
[Feature] Add Footer

### PR DESCRIPTION
**Pull request by gertie-sheshe**
## Description

Add Footer - include Newsletter, Description, Stay in Touch and Links

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots
Screenshot 2022-05-05 at 09 47 28
Screenshot 2022-05-05 at 09 47 57
Screenshot 2022-05-05 at 09 48 27

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: apps/codeforafrica/src/config/index.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import facebook from "@/codeforafrica/assets/images/facebook.png";
```
Comment (kilemensi) [inline apps/codeforafrica/src/config/index.js:1]: Instead of using `config`, use `lib/index.js` to return expected props. We then only have one place to change when we switch to the back-end CMS. Otherwise, `config` always end up with unused imported assets, etc.

File: apps/codeforafrica/src/components/StayInTouch/StayInTouch.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
+import { A } from "@commons-ui/core";
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/StayInTouch.js:1]: Should be using the `Link` component from `@commons-ui/next`, no?

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/codeforafrica/src/components/Footer/Footer.js [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
 
+import FooterDescription from "./FooterDescription";
+
+import FooterLinks from "@/codeforafrica/components/FooterLinks";
 import NewsletterSubscription from "@/codeforafrica/components/NewsletterSubscription";
+import StayInTouch from "@/codeforafrica/components/StayInTouch";
+import { socialMedia, footerLinks, footer } from "@/codeforafrica/config";
 
-const FooterRoot = styled(Box)(
+const FooterRoot = styled("footer")(
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply apps/codeforafrica/src/components/Footer/Footer.js]: Got it

File: apps/codeforafrica/src/components/FooterLinks/FooterLinks.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  }
+  return (
+    <ListRoot>
+      {footerLinks.main.map((item) => (
+        <ListItemLinks key={item.name}>
+          <LinkRoot href={item.href}>
+            <Typography variant="h5">{item.name}</Typography>
+          </LinkRoot>
+        </ListItemLinks>
+      ))}
```
[the remark this answers is not in the document]
Comment (gertie-sheshe) [reply apps/codeforafrica/src/components/FooterLinks/FooterLinks.js:61]: Hehehehe...ati what? @kilemensi

**Review by nerdlet — APPROVED** [review]

**Review by kilemensi — COMMENTED** [review]
👍🏽 


---

**UI**
1. In **mobile**, the order of components don't seem to match the designs: Newsletter component is first in the designs but it's last in the implementation.
2. Stay in touch icon images don't load for me
Screenshot from 2022-05-12 09-16-43

**CODE**
I think the code is still a bit too hard to follow. For example [this](https://github.com/CodeForAfrica/ui/pull/60/files#diff-1ae86086380727cfc2ac98229bf82807891eb9fbc3b5bec9389999fc88668c4bR43-R63) part of `FooterLinks`. We have `ListRoot` (a `ul`) but immediately followed by `Grid item` (aside, `Grid item` **_must_** always be inside `Grid container`) then followed by `SecondaryList` (a `div`). May be not a requirement but once you go `ul`, one expects _all_ immediate children to be `li`; if you go `Grid container`, one expects to see `Grid item` children, etc. One of simplest designs would be to make `FooterNavList` accept `children` that will be rendered [last](https://github.com/CodeForAfrica/ui/blob/c3f98816c118239900db64a8c1b1262bffd35745/apps/codeforafrica/src/components/FooterNavList/index.js#L43-L44). This way you can add extra `NavListItem`s that are not part of _main_ menu without having to create any extra components/wrappers.

File: apps/codeforafrica/src/components/FooterLinks/FooterLinks.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+}));
+function FooterLinks({ footerLinks, menu }) {
+  if (!footerLinks) {
+    return null;
+  }
+  return (
+    <ListRoot>
+      <FooterNavList menu={menu}>
+        {footerLinks.secondary.map((item) => (
+          <ListItemRoot key={item.name}>
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/FooterLinks/FooterLinks.js:44]: `NavList` already has [`NavListItem`](https://github.com/CodeForAfrica/ui/tree/main/apps/codeforafrica/src/components/NavListItem). Why not use that?

File: apps/codeforafrica/src/components/StayInTouch/StayInTouch.js [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+  name: "StayInTouchLinks",
+})(({ theme }) => ({
+  "& > a": {
+    display: "inline-block",
+  },
+  justifyContent: "center",
+  [theme.breakpoints.up("md")]: {
+    justifyContent: "flex-start",
+    width: "auto",
+  },
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/StayInTouch.js:20]: All of this can be in `sx` prop on `Grid item`... Feels "foreign" to see `LinksRoot item`

File: apps/codeforafrica/src/components/FooterLinks/FooterLinks.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            <LinkRoot href={item.href}>
+              <Typography variant="p2">{item.name}</Typography>
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/FooterLinks/FooterLinks.js:46]: Single `Link` can replace the use of `List` and `Typography`

**Review by kilemensi — COMMENTED** [review]
👍🏽 ... getting close

---------------

We should download SVG icons from the icons [page](https://www.figma.com/file/MbCygAaNhIjgJOHrQZpZ2U/CfA---CfA-site?node-id=3%3A5) in Figma and save them (keeping original names) in `public/icons` folder or something. This way we don't end up with duplicates of the same icon. (Twitter icon is used in multiple components for example).

File: apps/codeforafrica/src/components/FooterLinks/FooterLinks.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
+import { Link } from "@commons-ui/next";
+import { List } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import PropTypes from "prop-types";
+import React from "react";
+
+import FooterNavList from "@/codeforafrica/components/FooterNavList";
+import NavListItem from "@/codeforafrica/components/NavListItem";
+
+const ListRoot = styled(List)(({ theme: { breakpoints, typography } }) => ({
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/FooterLinks/FooterLinks.js:10]: Why `List`? Thought it will be a `nav`

File: apps/codeforafrica/src/components/StayInTouch/StayInTouch.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+      >
+        {socialMedia.map((media) => (
+          <Link
+            sx={{
+              display: "inline-block",
+              padding: 0,
+              paddingRight: "0.625rem",
+            }}
+            key={media.url}
+            href={media.url}
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/StayInTouch.js:64]: These links seem to open on same tab. Shouldn't they open on [new](https://github.com/CodeForAfrica/ui/blob/90d622b6b82fb440e2f10515e34cc82bd1b20951/packages/commons-ui-next/src/Link/Link.js#L92-L102) tab?

File: apps/codeforafrica/src/components/Footer/FooterDescription.js [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+import cfaIcon from "@/codeforafrica/assets/images/CfAlogoBW.png";
+
+function FooterDescription({ description }) {
+  if (!description) {
+    return null;
+  }
+
+  return (
+    <>
+      <Image src={cfaIcon} alt="" />
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/Footer/FooterDescription.js:15]: We should at least use `ImageButton` or `Link` to make this link to the homepage.

File: apps/codeforafrica/src/components/StayInTouch/Title.js [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+    },
+  })
+);
+
+function Title({ title }) {
+  return <TypographyRoot>{title}</TypographyRoot>;
+}
+
+Title.propTypes = {
+  title: PropTypes.string,
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/Title.js:22]:
```suggestion
  children: PropTypes.string,
```
I think `<Title>Goes here</Title>` is better than `<Title tile="Goes here" />`. I also think `<Typography variant="footerCap">` captures most of the styles.

File: apps/codeforafrica/src/components/StayInTouch/Title.js [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    textTransform: "uppercase",
+    fontWeight: "bold",
+    marginBottom: typography.pxToRem(23),
+    [breakpoints.up("md")]: {
+      marginBottom: 0,
+    },
+  })
+);
+
+function Title({ title }) {
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/Title.js:17]: Lets give it a better name and make it a standalone component since it can also be used as title in ShareBar... Something like `StayInTouchTitle`, `ConnectTitle`, `SocialLinksTitle`, etc.

File: apps/codeforafrica/src/components/Footer/Footer.js [hunk 13]
```diff
@@ -23,24 +25,38 @@ const FooterRoot = styled(Box)(
...
             xs={12}
-            md={3}
-            sx={{
-              order: { xs: 2, md: 1 },
-            }}
+            md={4}
+            sx={{ order: { xs: 2, md: 1 } }}
           >
-            <FooterNavList menu={menu} />
+            <FooterLinks footerLinks={footerLinks} menu={menu} />
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/Footer/Footer.js:56]: May be `additionalLinks`, `extraLinks`, etc.? `footerLinks` in `FooterLink` doesn't sound right.

File: apps/codeforafrica/src/components/FooterLinks/FooterLinks.js [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+  marginTop: typography.pxToRem(85),
+  "& li:nth-last-child(2)": {
+    marginTop: typography.pxToRem(20),
+  },
+  [breakpoints.up("md")]: {
+    marginTop: 0,
+  },
+}));
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/FooterLinks/FooterLinks.js:18]:
```suggestion
}));

```

File: apps/codeforafrica/src/components/Footer/FooterDescription.js [hunk 15]
```diff
@@ -0,0 +1,34 @@
...
+import PropTypes from "prop-types";
+import React from "react";
+
+import cfaIcon from "@/codeforafrica/assets/images/CfAlogoBW.png";
+
+function FooterDescription({ description }) {
+  if (!description) {
+    return null;
+  }
+
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/Footer/FooterDescription.js:12]:
```suggestion
```

File: apps/codeforafrica/src/components/StayInTouch/StayInTouch.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+}));
+
+const StayInTouch = React.forwardRef(function StayInTouch(
+  { children, socialMedia, title, ...props },
+  ref
+) {
+  if (!socialMedia?.length) {
+    return null;
+  }
+
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/StayInTouch.js:25]:
```suggestion
```

File: apps/codeforafrica/src/components/StayInTouch/StayInTouch.js [hunk 17]
```diff
@@ -0,0 +1,91 @@
...
+          sx={{ textAlign: { xs: "center", md: "left" } }}
+        >
+          <Title title="Stay in touch:" />
+        </Grid>
+      )}
+      <Grid
+        sx={{
+          width: { md: "auto" },
+          justifyContent: { md: "flex-start", xs: "center" },
+        }}
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/StayInTouch/StayInTouch.js:50]: We don't need `sx` I think... Grid props are responsive I think i.e. `<Grid justifyContent={{ xs: "center", md: "flex-start"` should work, no?

**Review by kilemensi — APPROVED** [review]
🚀 

-----

**BONUS**
Instead of `apps/codeforafrica/public/icons/Type=facebook, Size=32, Color=White.svg`, we can name them to `apps/codeforafrica/public/icons/Type=facebook, Size=32, Color=currentColor.svg` and edit the SVG replacing `white` with `currentColor`. This way, a single icon can be used everywhen by changing `color` of parent component.
